robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py

```diff
@@ -0,0 +1,532 @@
+            self._robot.speech.speak(
+                "{}please look at me while I am looking for you".format(
+                    person_label+', ' if person_label else ''),
+                block=False)
+
+        start_time = rospy.Time.now()
+
+        head_goals = [kdl_conversions.VectorStamped(x=self._look_distance * math.cos(angle),
+                                                    y=self._look_distance * math.sin(angle),
+                                                    z=1.5,
+                                                    frame_id="/%s/base_link" % self._robot.robot_name)
+                      for angle in self._look_angles]
+
+        i = 0
+        attempts = 0
+
+        rate = rospy.Rate(2)
+        while not rospy.is_shutdown() and attempts < self._attempts and (rospy.Time.now() - start_time).to_sec() < self._search_timeout:
+            if self.preempt_requested():
+                return 'failed'
+
+            self._robot.head.look_at_point(head_goals[i])
+            i += 1
+
+            if i == len(head_goals):
+                i = 0
+                attempts += 1
+
+            self._robot.head.wait_for_motion_done()
+
+            found_people_ids = []
+            for _ in range(2):  # TODO: parametrize
+                self._image_data = self._robot.perception.get_rgb_depth_caminfo()
+                if self._image_data:
+                    success, found_ids = self._robot.ed.detect_people(*self._image_data)
+                else:
+                    rospy.logwarn("Could not get_rgb_depth_caminfo")
+                    success, found_ids = False, []
+                found_people_ids += found_ids
+
+            # Use only unique IDs in the odd case ED sees the same people twice
+            found_people = [self._robot.ed.get_entity(eid) for eid in set(found_people_ids)]
+
+            rospy.loginfo("Found {} people: {}".format(len(found_people), found_people))
+            found_people = [p for p in found_people if p]
+            rospy.loginfo("{} people remaining after None-check".format(len(found_people)))
+
+            robot_pose = self._robot.base.get_location()
+            # TODO: Check probable bug here
+            found_people = filter(lambda x: (x.pose.frame.p - robot_pose.frame.p).Norm() < self._look_distance,
+                                  found_people)
+
+            rospy.loginfo("{} people remaining after distance < {}-check".format(len(found_people), self._look_distance))
+
+            if self._properties:
+                for k, v in self._properties.items():
+                    found_people = filter(lambda x:
+                            self._check_person_property(x, k, v), found_people)
+                    rospy.loginfo("{} people remaining after {}={} check".format(len(found_people), k, v))
+
+            result_people = []
+
+            if self._query_entity_designator:
+                query_entity = self._query_entity_designator.resolve()
+                if query_entity:
```

Comment:
I would add an else statement which gives a warning that the entity could not be resolved.